tools/brl_checks.c

```diff
@@ -134,7 +134,7 @@ check_base(const char *tableList, const char *input, const char *expected,
 	int *outputPos = NULL;
 	int cursorPos = 0;
 	inbuf = malloc(sizeof(widechar) * inlen);
-	outbuf = malloc(sizeof(widechar) * outlen);
+	outbuf = malloc(sizeof(widechar) * (outlen + 1));
```

Comment:
Can you explain (in a comment) why the `+1`

Reply:
This fixes issue #1538 , which is caused by writing trailing '\0' for printing `outbuf`. In fact, for printing a C-format string with size `outlen`, we actually need to store the string with size `outlen + 1`.

You can see the printing code in `brl_checks.c:213`
```c
			outbuf[outlen] = 0;
			// ....
			print_widechars(outbuf, outlen);
```

Reply:
This fixes issue #1538 , which is caused by writing trailing '\0' for printing `outbuf`. In fact, for printing a C-format string with size `outlen`, we actually need to store the string with size `outlen + 1`.

You can see the printing code in `brl_checks.c:213`
```c
			outbuf[outlen] = 0;
			// ....
			print_widechars(outbuf, outlen);
```

Reply:
Should I add a comment in code?

Reply:
> Should I add a comment in code?

Yes, please